src/com/sfuentes/DigitalClockTest.java

```diff
@@ -0,0 +1,181 @@
+package com.sfuentes;
+
+import java.util.ArrayList;
+
+public class DigitalClockTest {
+  public static void main(String[] args) {
+
+    generateTime_IncorrectHours();
+    generateTime_IncorrectMinutes();
+    generateTime_IncorrectSeconds();
+    generateTime_IncorrectTime();
+
+    generateTime_firstHour();
+    generateTime_CorrectMinutes();
+    generateTime_CorrectSeconds();
+    generateTime_lastHour();
+  }
+
+  //  Test Incorrect Hours
+  public static void generateTime_IncorrectHours() {
+    DigitalClock times = new DigitalClock();
+    String expectedResult = "24:0:0";
+    ArrayList<String> actualResult = times.generateHours();
+    boolean i = true;
+
+    for (String result : actualResult) {
+      if (result.equals(expectedResult)) {
+        i = true;
+        break;
+      } else
+        i = false;
+    }
+    System.out.println("Test case 1: Test that the hour does not exist: \n\t" +
+        "Expected result: false \n\t" +
+        "Actual result: " + i);
+
+    System.out.println(!i ? "\tSUCCESSFUL TEST." : "\tTEST FAILED.");
+  }
+
+  public static void generateTime_IncorrectMinutes() {
+    DigitalClock times = new DigitalClock();
+    String expectedResult = "1:60:0";
+    ArrayList<String> actualResult = times.generateHours();
+    boolean i = true;
```

Comment:
The name of this variable is often used as a counter and here seems that is not being used for that. Consider changing this to a more appropriated name like found or something like that.

Reply:
Resolved.